src/lib/comms/sol-socket-impl-linux.c

```diff
@@ -152,31 +152,36 @@ sol_socket_linux_del(struct sol_socket *socket)
 }
 
 static ssize_t
-sol_socket_linux_recvmsg(struct sol_socket *socket, void *buf, size_t len, struct sol_network_link_addr *cliaddr)
+sol_socket_linux_recvmsg(struct sol_socket *socket, struct sol_buffer *buf,
+    struct sol_network_link_addr *cliaddr)
 {
     struct sol_socket_linux *s = (struct sol_socket_linux *)socket;
     uint8_t sockaddr[sizeof(struct sockaddr_in6)] = { };
-    struct iovec iov = { .iov_base = buf,
-                         .iov_len = len };
+    struct iovec iov = { };
     struct msghdr msg = {
         .msg_name = &sockaddr,
         .msg_namelen = sizeof(sockaddr),
         .msg_iov = &iov,
-        .msg_iovlen = 1
+        .msg_iovlen = 1,
     };
     ssize_t r;
 
-    /* When more types of sockets (not just datagram) are accepted,
-     * remember to err if !buf && type != DGRAM */
```

Comment:
Is this comment not applicable anymore? It looks more of a warning for the future coder...